doc/devel/coding_guide.rst

```diff
@@ -1,20 +1,74 @@
-.. _reviewers-guide:
+.. _pr-guidelines:
 
-********************
-Reviewers guideline
-********************
+***********************
+Pull request guidelines
+***********************
 
-.. _pull-request-checklist:
+Pull requests (PRs) are the mechanism for contributing to Matplotlibs code and
+documentation.
 
-Pull request checklist
-======================
+.. _pr-checklist-authors:
 
-Branch selection
-----------------
+Checklist for PR authors
+========================
+
+.. note::
+
+   * We value contributions from people with all levels of experience. In
+     particular if this is your first PR not everything has to be perfect.
+     We'll guide you through the PR process.
+   * Nevertheless, try to follow below guidelines as good as you can to
+     help make the PR process quick and smooth.
+   * Be patient with reviewers. We try our best to respond quickly, but we
+     have limited bandwidth. If there is no feedback within a couple of days,
+     please ping us by posting a comment to your PR.
+
+When making a PR, pay attention to:
+
+* :ref:`Target the master branch <pr-branch-selection>`.
+* Adhere to the :ref:`coding_guidelines`.
+* Update the :ref:`documentation <pr-documentation>` if necessary.
+* Aim at making the PR as "ready-to-go" as you can. This helps to speed up
+  the review process.
+* It is ok to open incomplete or work-in-progress PRs if you need help or
+  feedback from the developers.
```

Comment:
You may take advantage of GitHub's [draft pull requests](https://help.github.com/en/articles/about-pull-requests#draft-pull-requests) to signal this status more directly.